### ISSUE TITLE
Multi label image classification using convolutional neural network

### ISSUE THREAD

**Issue by suraj-deshmukh** [fchollet/keras#1585]
I am working on multi label image classification problem. The dataset is given on [this](http://lamda.nju.edu.cn/data_MIMLimage.ashx) link. I am using Convolutional Neural Network (CNN) with fully connected neural network (NN) at the end. I am using the keras package for the same. My question is how can I proceed to complete this task. Is there any specific method for this?

Right now I have implemented the CNN with NN and I am using sigmoid as activation function at the end and loss as binary crossentropy. And after testing on test set I am getting probabilities for different labels and by using different thresholds for each label I am converting them into zeros and ones and checking Matthews Correlation Coeficient [(MCC)](https://en.wikipedia.org/wiki/Matthews_correlation_coefficient) for each label separately.

Is my approach right or wrong?

**Comment by fchollet** [comment]
Sounds about right.

**Comment by suraj-deshmukh** [comment]
@fchollet 
ok thanks:)

Also I want to visualize the loss/acc vs epochs?  
Is there any in build method to do so?

**Comment by suraj-deshmukh** [comment]
@fchollet 
Actually i was thinking to add visualization utility to keras using matplotlib. It will plot the loss or acc vs epochs.

**Comment by suraj-deshmukh** [comment]
Hi,
   I am thinking to add plotting utility to keras so that user can get the
loss/acc vs epoch plot. If you want this then i well send you a pull
request after completing the code.



On Fri, Jan 29, 2016 at 12:09 PM, François Chollet <notifications@github.com
